### ISSUE TITLE
Deny `static mut` declarations entirely

### ISSUE THREAD

**Issue by pitaj** [rust-lang/rust-clippy#12896]
### What it does

The `static_mut_refs` rustc lint checks for shared or mutable references of mutable statics. This catches most incorrect uses, but doesn't necessarily accomplish three goal of moving the ecosystem away from `static mut` entirely. 

The `static_mut` clippy lint will check for any declarations of mutable statics, and recommend using an immutable static with a type with interior mutability instead.

### Advantage

Move the ecosystem away from `static mut`. May even catch some unsoundness issues.

### Drawbacks

Churn

### Example

```rust
static mut NUM: usize = 0;
static mut FOO: Thing = Thing::new();
```

Could be written as:

```rust
static NUM: AtomicUsize = AtomicUsize::new(0);
static FOO: Mutex<Thing> = Mutex::new(Thing::new());
// Or RwLock, LazyLock, SyncUnsafeCell, etc
```

**Comment by pitaj** [comment]
You also shouldn't claim an issue that's already assigned, without first attempting to contact the assignee.

**Comment by archief2910** [comment]
sorry @pitaj  it was by mistake that's why i unassigned the issue

**Comment by aaron-ang** [comment]
@lolbinarycat are you still working on this issue?

**Comment by pitaj** [comment]
@aaron-ang it's being worked on at https://github.com/rust-lang/rust-clippy/pull/13770

**Comment by lolbinarycat** [comment]
@aaron-ang not really, no.

**Comment by aaron-ang** [comment]
@pitaj can I confirm that #13770 is not related to this issue?

**Comment by pitaj** [comment]
Correct, #13770 is unrelated. I was confused earlier

**Comment by pitaj** [comment]
@aaron-ang you might want to read through #12914 before putting in any work.

**Comment by aaron-ang** [comment]
- References to `static_mut` have already been disallowed in `rustc`: https://github.com/rust-lang/rust/pull/117556 and https://github.com/rust-lang/rust/pull/124895
- Waiting for https://github.com/rust-lang/rust/issues/95439 to become stable so we can suggest it in the lint
- Rust community is planning to deprecate `static_mut` in the future: https://github.com/rust-lang/rfcs/pull/3560, pending `const_mut_refs` to be stabilised: https://github.com/rust-lang/rust/issues/57349
